src/updaters/java/pom-xml.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Version} from '../../version';
+import {BaseXml, findNodes} from '../base-xml';
+
+/**
+ * Updates version pom.xml files.
+ *
+ * If present it updates project.version element.
+ * If project.version is not present, it updates project.parent.version.
+ */
+export class PomXml extends BaseXml {
+  private readonly version: Version;
+
+  constructor(version: Version) {
+    super();
+    this.version = version;
+  }
+
+  protected updateDocument(document: Document): boolean {
+    const version = this.version.toString();
+
+    // Update project.version
+    const projectVersionNodes: Node[] = [
+      ...findNodes('/project/version', document),
+    ];
+    if (projectVersionNodes) {
+      // If found update, detect actual change
+      return PomXml.updateNodes(projectVersionNodes, version);
+    }
+
+    // Try updating project.parent.version
+    const parentVersionNodes: Node[] = [
+      ...findNodes('/project/parent/version', document),
+    ];
+    return PomXml.updateNodes(parentVersionNodes, version);
```

Comment:
Ok, it is matter of an opinion, but I can rework it so it will be handled by a plugin instead. I've just tried to provide out-of-the-box simple solution with minimal config, that would work with most common maven projects. You want something extra? Don't use maven strategy, use java strategy and with replaces driven by comments.

If we will rewrite this, I would suggest however to move this to separate PR, created after this is merged (that is, reducing the scope of this change and making it more managable).

Reply:
Ok, I'm fine with this as it is. I can see this working nicely for a multi-module project where all the modules are on the same version and that version is defined in the parent.

If a maintainer need different versions, then you would need to configure a multi-path manifest with a java-workspace plugin to make sure cross dependency versions are updated appropriately.